doc: Refactor "install boundary" section (#4557)
hashicorp/boundary: website/content/docs/install-boundary/initialize.mdx

---
layout: docs
page_title: Initialize Boundary
description: |-
  Creating your first login account
---

# Initialize Boundary

> Before you initialize Boundary, you should have [initialized a database](/boundary/docs/install-boundary/configure-controllers#initialize-the-database).

Reviewer: I think we could move this sentence below lines 12 & 13, and put it under a second level heading ## Requirements for visibility.

This document describes how to access Boundary for the first time and create the necessary resources to log in as a

> user in Boundary.

Reviewer: ```suggestion
user.
```

> ## Log in with Recovery KMS

Reviewer: ```suggestion
## Log in with recovery KMS
```

> In order to access a new Boundary instance for the first time, you will need to use the recovery KMS defined in the
> controller configuration file. Here is an example snippet of a configuration file that uses hard-coded AEAD keys.

Reviewer: ```suggestion
To access a new Boundary instance for the first time, you must use the recovery KMS defined in the
controller configuration file. The following example represents a configuration file that uses hard-coded AEAD keys:
```

> ~> **Note:** In a production environment, it is recommended to use a cloud provider's KMS such as [AWS KMS](/boundary/docs/configuration/kms/awskms)
> to manage the keys Boundary uses to encrypt sensitive information.

Reviewer: ```suggestion
<Note>

In a production environment, HashiCorp recommends that you use a cloud provider's KMS such as [AWS KMS](/boundary/docs/configuration/kms/awskms)
to manage the keys Boundary uses to encrypt sensitive information.

</Note>
```

Reviewer: This is a new format we are using for Markdown notes.

<truncated>
```

> The KMS with the `recovery` purpose is used to "recovery" Boundary, allowing you to authenticate to Boundary and managed
> it as a super user. It also allows you to authenticate from the CLI or from Terraform in order to manage BOundary
> without any generated resources.

Reviewer: ```suggestion
The KMS with the `recovery` purpose is used to "recover" Boundary, allowing you to authenticate to Boundary and manage
it as a super user. It also allows you to authenticate from the CLI or from Terraform to manage Boundary
without any generated resources.
```

Reviewer: I think you meant "recover" instead of "recovery" here, but please let me know if I misunderstood.

To use the recovery workflow on the CLI, you must pass the `-recovery-config <path_to_kms_recovery_config>` flag or set the environment
variable for `BOUNDARY_RECOVERY_CONFIG` for every command ran. Authentication takes place for every command

> ran when using the recovery workflow. `boundary authenticate` does not apply when using the recovery KMS.

Reviewer: ```suggestion
ran when using the recovery workflow. The `boundary authenticate` command does not apply when you use the recovery KMS.
```

Reviewer: I usually try not to start a sentence with a command, I think it can be a little difficult to read.

### Create roles to manage scopes

> The following describes the four baseline roles you will need to create in order to manage resources within the org and project
> scopes that were created. These roles are similar to the roles automatically created for you if generation had not been skipped during `boundary database init` when executed with the `-skip-initial-login-role-creation` flag, Declaring roles explicitly

Reviewer: ```suggestion
The following section describes the four baseline roles you must create to manage resources within the org and project
scopes that you created. These roles are similar to the roles Boundary automatically created for you if you did not skip generation using the `-skip-initial-login-role-creation` flag when you ran `boundary database init`. Declaring roles explicitly
```

### Log in as your new user

```bash
boundary authenticate password \
  -auth-method-id <auth_method_id>

> ```

Reviewer 2: I would probably keep this as `shell-session` instead of bash, since it's not a bash script. We tend to use shell-session for all generic executable code, unless it's bash-specific
